rascal-lsp/src/main/java/org/rascalmpl/vscode/lsp/BaseLanguageServer.java

```diff
@@ -278,53 +280,14 @@ private static PathConfig findPathConfig(ISourceLocation path) throws IOExceptio
         }
 
         @Override
-        public CompletableFuture<ProjectLibFolder[]> supplyProjectLibFolders(URIParameter projectFolder) {
+        public CompletableFuture<IConstructor> supplyPathConfig(URIParameter projectFolder) {
             return CompletableFuture.supplyAsync(() -> {
                 try {
-                    if (projectFolder.getUri() == null) {
-                        return new ProjectLibFolder[0];
-                    }
-                    PathConfig pcfg = findPathConfig(projectFolder.getLocation());
-                    // in the future we want to have lib paths that are pointing to something stable
-                    // but right now our lib path are relative to the project they belong too
-                    // so for now we re-do the mapping here.
-                    // we check the javaCompilerPath RASCAL.MF files, and see if they are present in the lib path
-                    // if that is the case, we add it to this mapping
-
-                    var result = new ArrayList<ProjectLibFolder>();
-                    var seen = new HashSet<String>();
-                    seen.add("");
-                    for (IValue vl : pcfg.getLibs()) {
-                        var l = (ISourceLocation) vl;
-                        if (l.getScheme().equals("lib")) {
-                            var libName = l.getAuthority();
-                            if (!seen.contains(libName)) {
-                                // we have to lookup the corresponding jar/folder on the class loaders
-                                pcfg.getClassloaders().stream()
-                                    .map(ISourceLocation.class::cast)
-                                    .map(RascalManifest::jarify)
-                                    .filter(p -> !p.getScheme().equals("lib"))
-                                    .filter(p -> new RascalManifest().getProjectName(p).equals(libName))
-                                    .findFirst()
-                                    .ifPresent(foundPath -> {
-                                        result.add(new ProjectLibFolder(libName, foundPath));
-                                        seen.add(libName);
-                                    });
-                            }
-                        } else {
-                            l = RascalManifest.jarify(l);
-                            var libName = new RascalManifest().getProjectName(l);
-                            if (!seen.contains(libName)) {
-                                result.add(new ProjectLibFolder(libName, l));
-                                seen.add(libName);
-                            }
-                        }
-                    }
-                    return result.toArray(new ProjectLibFolder[0]);
+                    return PathConfig.fromSourceProjectMemberRascalManifest(projectFolder.getLocation(), RascalConfigMode.INTERPETER).asConstructor();
```

Comment:
Could we return the map of keyword parameters here as a generic object? That way the code is 100% robust against changes in PathConfig.

Reply:
ok, changed that.